### DISCUSSION TITLE
✅ 🚀 Feature Request: Taxonomies as template type in Loop Grid + Loop Carousel, to Loop over taxonomies instead posts

### DISCUSSION THREAD

**Discussion by ayoubkhan558** [elementor/elementor#21372] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

I want to Loop over WooCommerce product categories, But I find that there was no taxonomy template type in loop grid options.
![219390427-3738d92e-9441-40c8-8fc5-a93ffa3c9857](https://github.com/elementor/elementor/assets/99708427/93f788f8-31b9-4d20-a780-14927aa9203e)

### Describe the solution you'd like

Please add WordPress taxonomies ( category of post/product, tags, custom taxonomies etc) as template type so that If we want to loop over them, it's easy for us.
It should not take too much time. It's a great option to have. We don't have to use any third-party option to loop over taxonomies.

### Describe alternatives you've considered

I am using ThePlus Addons as for now.
https://theplusaddons.com/pluslisting/dynamic-listing/
https://theplusaddons.com/pluslisting/dynamic-category/

### Additional context
After adding this feature, It will show taxonomies in the template menu as well.  and we will be able to loop over taxonomies instead of posts
![taxonomy loop grid builder](https://github.com/elementor/elementor/assets/99708427/eb6c9799-55ea-4d0a-806d-3d4cf7b48aca)

I will be able to loop over custom taxonomy like "Skill" and create a grid of skills 
![front-end-technologies](https://github.com/elementor/elementor/assets/99708427/ce5b0e9c-7b80-4d82-9983-bf96473a9024)

**Comment by Michaelgimii** [comment]
I have raised this issue many times. Currently, in the loop grid/carousel widget, in the Choose template type section, only two options, products and posts, are displayed. Therefore, if we have a custom post type, it will not be displayed in this section. Instead, we have to select it in the query tab > query!
I love Elementor. But to be honest, the ux in the two loop widgets is extremely weak. very weak. I think the image below is something that would be great and basic for two loop widgets.

![loop](https://user-images.githubusercontent.com/88034902/219599454-8f944ef6-d675-40df-927f-97db8dd9f690.png)

**type**

- posts
- terms (product category)
- users

**post type**

- posts
- products
- home (Custom post type)
- pages
- media

Dear Elementor team @avivu , @arielk , don't you think this is more logical and better than the current situation? Even if you don't agree with my suggestion, I'd be happy to hear your opinion❤.

**Reply by ayoubkhan558** [reply]
Thanks for raising the issue and suggesting a solution.

**Comment by formulaclick** [comment]
+1 This would be great to get rid of crocoblock!!!

**Comment by Joliamendi** [comment]
+1000
Please add this feature to both loop grid and loop carousel widgets.

**Comment by Michaelgimii** [comment]
@ayoubkhan558, Please correct the title. in **Loop Grid + Loop Carousel**

**Reply by ayoubkhan558** [reply]
Thanks :) Corrected

**Comment by tomaszsoin** [comment]
+1

**Comment by screwdycrow** [comment]
this is something that the new loop builder should be able to do. Although i do understand that querying posts and querying taxonomy terms, so i have a feeling that it might not be as straightforward as one  may think. Nevertheless making an infrastructure to be able to query taxonomy terms in the loop grid and  loop carousel should be a priority and something that many people would use, especially on eshops.

**Comment by pixelloop** [comment]
This is something I would have expected to already exist in the new loop builder. 

I use this a lot with some other page builders, and it's very handy.

At the moment, I'm having to take a different approach with Elementor which means using more plugins and bloat.

**Comment by pbaddock** [comment]
+1 - this would be fantastic.

**Comment by jfernandezr** [comment]
+1, absolutely needed

**Comment by Lonsdale201** [comment]
+1 :)

**Comment by maxjutras** [comment]
+1! Would be sooo useful!

**Comment by alloutsidethebox** [comment]
Yes please, this would be soooo useful devs make it so.

**Comment by Zegklitz** [comment]
+1

**Comment by fabigrafo** [comment]
This would be so helpful!

**Comment by ayoubkhan558** [comment]
?

**Comment by ayoubkhan558** [comment]
Why elementor is so slow in responding to issues, or creating new features?
Learn something from the Breakdance page builder.

**Reply by pixelloop** [reply]
Unfortunately, Elementor's priority and business model is enticing new customers with fancy but unnecessary features like AI rather than providing more essential features for developers. 

It's a shame because most of the other page builders are the other way around.

**Reply by lensbreak** [reply]
so true, since years elementor management seems like having only their ideas in their heads and not listening to their PAYING customers. unfortunately, using another pagebuilder in 2023 is a bit risky if you use it for a serious client project.

**Reply by pixelloop** [reply]
Agreed, and people like @nicholaszein will downvote anything even remotely critical of Elementor development. There are so many babies in the Elementor space. Try putting any sort of critique on Reddit, and you will be downvoted to death.

Nicholas, We are only critiquing Elementor in the hope that it can be better. Many of us come from backgrounds of working with other page builders, which have tools from day 1 that Elementor is still lacking years later. Essential stuff, you know, not just shiny bells and whistles.

**Reply by ayoubkhan558** [reply]
#22945 created a discussion for this.  
Please upvote there, discuss there.

**Reply by ayoubkhan558** [reply]
Please try to reference simple features which are still not added by elementor, but request long time ago. Maybe someday elementor team look into it.

**Reply by nicholaszein** [reply]
@pixelloop, @lensbreak, and @ayoubkhan558 thank you for your feedback.

Let me start by saying that reacting to a comment does not mean down voting it. I just didn't agree with the critique that was made because it wasn't the truth.

We are constantly watching for feature requests. In fact, this is one that is a contender for our Roadmap.

But posting a feature request does not guarantee that it will be immediately included in our Roadmap. There is a process of research, a queue to follow and lots of things to consider. 

It's unfair to assume Elementor is acting against the <ins>entire</ins> community. Our interests are in fact very aligned, but it takes time to make things right. We could easily have scraped the entire plugin, started it over, given it another name, and charged everyone again if they wished the new shiny features that other builders don't have, while praising our was better. Instead we choose to embrace our community and slowly but responsibly restructure our plugin to be able to add new features and account for backward compatibility. This takes time, and requires patience and planing. And we want to make it right. 

It's important also to note that this space is, in fact, one that is self-sustained. The idea is to give you the tools to be able to discuss and vote (using the upvote button) for you favorite suggestions, without us needing to babysit every single post and comment. This help us understand our users needs, and plan ahead accordingly. This also means we do not need to answer every single feature request. This is our GitHub forum, we receive notifications about every single new thread, and we visit them all every day. We are aware of all new issues and discussions. We are listening. And to listen sometimes we need to be quiet. But this does not mean we don't care. It's the opposite. We care, and a lot, and we want to continue both improving, and innovating, and this will take time. 

We appreciate your patience as we take the time to review your request thoroughly. We aim to ensure all decisions align with our mission to provide Web Creators with the tools they need to build their dream websites.

Kind regards

**Reply by ayoubkhan558** [reply]
Other page builders add what is requested by developers, but you add what you think is best?
Then who is on the right path?
You encourage to add plugins for even a simple functionality.  But other builders are adding it as their core functionality?

Conditional visibility was requested in 2018, You have 400+ Developers, It's 2023 and still we have to look for a third-party plugin to achieve this, instead, you should have added this as core functionality. 

So who is on the right path, sir?

We are not elementor critics or enemies in fact we used to love it, so want to improve it. But the only problem is that elementor team prorioties.

**Reply by ayoubkhan558** [reply]
We can suggest better what is actually needed, What's useful to developers. 
It's your duty to add that instead of adding useless features/

**Reply by nicholaszein** [reply]
@ayoubkhan558 sorry, perhaps I was not clear enough. Let me try to rephrase what I meant:

1. Elementor IS indeed using your feedback to plan for our Roadmap. However, with such a huge user base, we are focusing on delivering what is most asked by users, based on feedback, but also based on R&D and our vision for our products. None of these criteria negates the other.

2. What the majority of users request, might not be what individual users think is the most requested features. Please keep that in mind.

3. Some of the requests, like the Display Conditions you mentioned require profound changes to our codebase for a better, more efficient, and long lasting solution. Changing our codebase requires planing and responsibility. We cannot introduce breaking changes without giving time for everyone, including 3rd-party developers and users, to gradually adapt to them.

4. Just because another builder or a 3rd-party add-on came up with a solution, doesn't mean it's the best solution if we want a reliable, stable, and long-lasting solution that can be safely built-upon, which is in the interest of our users.

5. Responsible changes take time.

6. Elementor has multiple products, and multiple teams working on various features, solutions, and departments that usually happen way ahead (sometimes months) of their release. If a Product or feature is being worked on, it doesn't mean we stopped working on anything else. Including following and acting on all the discussions and issues posted on GitHub.

7. We will always focus on working for our entire user base. That includes you, but other people too. We can't appease everyone's expectations all the time, that would be unrealistic. And that's OK. We want to be inclusive. This means doing our best to work for solutions that will benefit the majority of our users. This also means that our Roadmap is not planned solely on who screams louder, but rather on how many people need something, and how we can deliver what they need in a responsible manner. Many times, it WILL take time to accomplish this.

All we ask if for your patience, and we appreciate that you care enough to share your feedback! Thank you for speaking out! 🙏

**Reply by ayoubkhan558** [reply]
Was AI Image generator more needed/request or Conditional widget/sections display?
I can't see anywhere that people have asked about AI Image generators more than Conditional widgets/sections, AI is just 7 months old, and this was requested 5 years ago. and still missing, It's requested by thousands. 

Any enough discussion, Good luck Sir. We can't suggest you more.

**Reply by pixelloop** [reply]
65 upvotes and many replies, and it's only a contender for the roadmap? Now I'm shocked! 

It's been obvious for a long time that Elementor prioritises their own ideas that will sell more copies over basic features that should have existed years ago. 

To be clear, I don't expect things to be instantly added to the roadmap. However, professional developers should have thought of this feature many years ago, so the fact it takes this long is surprising. 

I only use Elementor sparingly because I consider BeaverBuilder to be a better solution as they have had features like this for years and properly considered what's essential vs what's nice to have, the opposite mantra of Elementor, unfortunately. 

I'm open to being proven wrong, but I'm not holding my breath.

**Comment by lensbreak** [comment]
+1 very much needed. especially with the new ACF version where we can make custom post types. for ecommerce we would have a very powerful setup then with Elementor Pro + ACF. @rami-elementor make it ready in the next release please!

**Comment by arekuki** [comment]
+1

**Comment by fabigrafo** [comment]
Elementor!! 👀

**Comment by ghost** [comment]
+1

**Comment by ayoubkhan558** [comment]
+1

**Comment by alsage13** [comment]
+1,000!!!!!!!!!!

**Comment by elhamshari** [comment]
+1

**Comment by ofyalcin** [comment]
+1

**Comment by nicholaszein** [comment]
Hello, please instead of commenting "+1", use the upvote button on the initial comment.

**Reply by pixelloop** [reply]
People only tend to use +1 out of frustration and as a way of bumping the thread. ayoubkhan558's comments are valid. AI wasn't requested more than this feature or conditional display but it's obvious Elementor saw a shiny new bell and decided it would sell more copies than an essential but less sexy developer feature.

**Comment by creativeart** [comment]
I hope it will be solved. A few days ago, my client asked to me to display custom taxonomy on a page. I was believing it was possible with the grid loop widget but it seems no :(
The university server use a old PHP version (7.2), so I cannot use Jet Engine. Do you know a plugin compatible with PHP 7.2 to display a grid of any taxonomy we want by the way ?
Thank you.

**Reply by lior10006** [reply]
https://dudaster.com/ecs-pro/

**Reply by fabigrafo** [reply]
Haven't we already said that we don't like other plugins?

**Reply by nicholaszein** [reply]
Elementor requires PHP 7.4 as the minimum version. Please contact your TI department and ask them to update to at least 7.4.

**Comment by drewjohnstone** [comment]
+1

**Comment by Zegklitz** [comment]
+1

**Comment by nicholaszein** [comment]
Hello everyone, I see more "+1s" than additiomal upvotes to the request in question.

Please don't forget to actually vote for the request by clicking the ⬆️ up arrow in the first comment. We use this to decide which features to add next. Commenting "+1" does not help.

**Reply by ayoubkhan558** [reply]
Please 🙏 do a favor count +1 as up votes 😉

**Reply by nicholaszein** [reply]
Not the intended purpose of having Feature Requests as Discussions. 

The idea is to allow people to vote and count their votes. If they comment "+1" it negates the whole purpose of counting votes. 

We have at least 800+ feature requests to go through. If you wish us to go faster we need your help to actually collaborate with the voting process. 

Clicking a button instead of commenting "+1" shouldn't be a chore or a problem. It should actually be easier to accomplish.

**Comment by nicholaszein** [comment]
This request has Duplicates:

- #22054 
- #22592

**Reply by growthsquad** [reply]
I'm new here, came to post a similar feature request. I need this feature as well to create a simple category grid of a custom post type. Just out of curiosity, how many upvotes does it take for the request to be added to "Planning" on the roadmap? I went though all 32 pages of feature requests and this is the only request with 87 upvotes, 101 if you add in the duplicate upvotes. 

Honestly, I was surprised it's not already a feature in the Loop Grid Widget, the LGW is the best tool for this job. By the way, the LGW is a great addition to Elementor, and much appreciated. With this highly requested feature added, it would be even better!

In my opinion, great products, solve real problems. Most designers/developers when faced with the same issue over and over again, tend to no longer ask for feature requests, they move on to other solutions. I was a Joomla guy for 14 years, moved to Wordpress because Joomla stopped solving problems. Elementor was the #1 reason I made the switch.

I do hope this is considered soon, currently I have to create a manual grid of categories with lots of containers to get a clean 4 column layout. It's quite a bit of work, especially when you add a new categories and need to re-alphabetize the entire grid. Not fun.

**Comment by Qadir9813** [comment]
much needed

**Comment by Zubair-dvlpr** [comment]
elementor solve this problem, we need it.

**Comment by usama720** [comment]
waiting for this feature

**Comment by bilalbhatti56** [comment]
elementor team please focus here.  @nicholaszein

**Comment by malay009-star** [comment]
Please add these features to both the loop grid and loop carousel widgets.

**Comment by trofimartya** [comment]
+1

**Comment by Archicoders-Rein** [comment]
+1

**Reply by ayoubkhan558** [reply]
Please vote for the request by clicking the ⬆️ up arrow in the first comment.

**Comment by jregist** [comment]
I badly need this functionality.

**Reply by ayoubkhan558** [reply]
Please vote for the request by clicking the ⬆️ up arrow in the first comment.

**Comment by adam-gre** [comment]
Here's a snippet from my functions.php that I wrote to get around this. It doesn't use loop grids but does get subcategories of current parent category so could still be useful for some

```
// Category navigation
function m8_get_categories() {

	$taxonomy     = 'product_cat';
	$orderby      = 'name';  
	$show_count   = 0;      // 1 for yes, 0 for no
	$pad_counts   = 0;      // 1 for yes, 0 for no
	$hierarchical = 1;      // 1 for yes, 0 for no  
	$title        = '';  
	$empty        = 0;
  
	$args = array(
		   'taxonomy'     => $taxonomy,
		   'orderby'      => $orderby,
		   'show_count'   => $show_count,
		   'pad_counts'   => $pad_counts,
		   'hierarchical' => $hierarchical,
		   'title_li'     => $title,
		   'hide_empty'   => $empty
	);

   $all_categories = get_categories( $args );
//    echo print_r(get_queried_object());
   
   foreach ($all_categories as $cat) {
	  if($cat->category_parent == 0) {
		  $category_id = $cat->term_id;       
		//   echo '<br /><a href="'. get_term_link($cat->slug, 'product_cat') .'">'. $cat->name .'</a>';
  
		  $args2 = array(
				  'taxonomy'     => $taxonomy,
				  'child_of'     => 0,
				  'parent'       => get_queried_object_id(),
				  'orderby'      => $orderby,
				  'show_count'   => $show_count,
				  'pad_counts'   => $pad_counts,
				  'hierarchical' => $hierarchical,
				  'title_li'     => $title,
				  'hide_empty'   => $empty
		  );
		  $sub_cats = get_categories( $args2 );
		  
		//   echo print_r($sub_cats);	

		$foreach_bit = "";

		// TODO - Fix URL below
		if ($sub_cats) {
			foreach($sub_cats as $sub_category) {
				$foreach_bit .= "<a class='m8_filter_link' href='" . get_category_link( get_queried_object_id() ) . "?group_type=" . $sub_category->term_id . "'><div class='m8_category'>" . $sub_category->name . "</div></a>";
			}
		}
		

		$html_result = "<div class='m8_flex'>" . $foreach_bit . "</div>";
			return $html_result;
	  }       
  }
}
add_shortcode('m8_categories', 'm8_get_categories');
```

**Comment by Isabellasen** [comment]
It would be really nice to have this option. We are a big web agency who all are missing it.

**Reply by ayoubkhan558** [reply]
Please vote for the request by clicking the ⬆️ up arrow in the first comment.

**Comment by awebguy99** [comment]
Yes, it really needs this.

**Comment by jsayubi** [comment]
We need this!

**Reply by pixelloop** [reply]
We will just get pointless AI features that we have lived happily without for years, unfortunately. Essential and basic template controls are a low priority.

**Comment by 79ho3ein** [comment]
This feature has recently been added to the roadmap.

https://elementor.com/roadmap/

**Reply by pixelloop** [reply]
Hopefully custom taxonomies are supported as the entry only states categories and tags.

**Reply by ayoubkhan558** [reply]
@nicholaszein

**Display posts and products in your Loop Grids based on tags or categories.**
This is what the roadmap says, I think they are getting it wrong, As this discussion was to add a feature that loops over taxonomies, not loop products and posts.  

https://elementor.com/category/roadmap/planning/

**Reply by nicholaszein** [reply]
You are correct @ayoubkhan558!

We are going to fix this typo! 😉 

Thanks for bringing that to our attention! 🙏

**Reply by Joliamendi** [reply]
Hi @nicholaszein,

will the taxonomy query be added only for the Loop Grid widget? Only Loop Grid is mentioned in the roadmap. How about loop carousel?

**Reply by nicholaszein** [reply]
Hi @Joliamendi!

The feature will be added to the Query component, so the answer is yes, even if it takes a few versions in between to sort all the differences out. 😉

**Comment by komizutama** [comment]
If I could upvote this a million times, I would.

**Comment by nicholaszein** [comment]
Hey everyone,

If you haven't seen already, **this feature is currently being tested in version 3.21-beta**. We expect to release the final version on April 15th, next week!

We would love to know your feedback about it in the [beta discussion](https://github.com/orgs/elementor/discussions/25589)! 💪 

Thank you all for your contribution! 🙏

**Comment by awebguy99** [comment]
@nicholaszein Thanks for the update, that's great news! I look forward to trying it out.

**Comment by nicholaszein** [comment]
Hey @everyone!

We have great news! 🙌 

### 📢 We're happy to announce that this feature was released with Elementor Pro v3.21.0! 🥳 

You can now use the Loop Grid and Loop Carousel to display Taxonomies, together with their custom meta such as ACF images, using your own designs!

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
